### DISCUSSION TITLE
Buiding errors arising from the thridparty directory

### DISCUSSION THREAD

**Discussion by Zarca** [v6d-io/v6d#156] [Q&A] [answered]
Help,
I'm trying to build the libvineyard from source,
the error message "there is no CMAKELists.txt in thirdparty/nlohmann"   occurs while i'm using the cmake tool-kit to build it,
And i also notice that all the direcotries in the thirdparty(like uri, pybind)   seem to lack the CMAKELists.txt 
Any suggestion for it?

**Comment by sighingnow** [comment, marked as answer]
Have you run

```bash
git submodule update --init
```

after git clone?

**Comment by sighingnow** [comment]
You could find more detail about how to build from source at https://v6d.io/notes/install.html#install-from-source

**Comment by Zarca** [comment]
Yes,i did not use the submodule update cause i just dowloaded it's source code(...tar.gz) directly from the  release page.
It is now okay after using git to fork a clone from the git repo and do the submodule update .Thanks for it

**Comment by sighingnow** [comment]
Well done. Free to reopen if you have any further questions. It seems impossible to include submodules in auto-generated source archive. We may upload a manually packaged one later for every release.
